spec/unit/hanami/config_spec.rb

```diff
@@ -11,8 +11,8 @@
     end
 
     before do
-      config.environment :production do |c|
-        c.logger.level = :info__set_for_production_env
+      config.environment :production do |env|
+        env.logger.level = :info__set_for_production_env
```

Comment:
@timriley This is confusing 😕 .

All the settings (like `logger`) are usually accessible via `config`.
Why adding more concepts like `env`, as yielded object?

Is it possible to keep `config`?

Reply:
This is less a "new concept", just an explicit local variable name. We can call it env_config if you like. Or c. Calling it config causes variable shadowing so I was looking to avoid that. 

As I explained in the commit message, we want to encourage use of the yielded object (at least with the current implementation) because if you just used the bareword `config` inside the block, it would actually return the _internal_ Dry::Configurable::Config object (because this block is being instance_evaled), which would be surprising behavior for someone expecting config there to be the Hanami::Config object. To access that Hanami::Config object inside the block without using the block argument, you have to use self. 

The other option here is to create a custom evaluation context for that block that wraps the Hanami::Config instance and returns _that_ as config, but I felt that would be adding more complication than we might like, especially since we have the block argument we can leverage.

Reply:
To put it another way, usage of the bareword `config` inside the block will only work for the accessors defined on the class using `setting`, because they're available as methods on the internal Dry::Configurable::Config object. Any methods declared on Hanami::Config will _not_ be available on the bareword config inside this block. That will be surprising and confusing behavior.  You can use `self` instead of course, but that doesn't feel good either. So by recommending the use of the block arg, we avoid this issue entirely.

Reply:
@timriley OK, so which type is `env`? 🙂

Reply:
@jodosha its in the docs I added ;)

```
@yieldparam whatever_name_you_prefer [self]
```

So it's the Hanami::Config instance. This is just the built-in behavior of instance_eval - it yields its receiver to the block. 

Happy to adjust the specs and docs to use a different name for this block param.

Reply:
@timriley `cfg`? or simply `c`?

Reply:
@jodosha I'll just call it `c`, then I'll merge this :)